services:
  kong:
    image: kong/kong-gateway:3.10.0.2
    depends_on:
      - auth-service
    volumes:
      - ./kong/kong.yml:/usr/local/kong/declarative/kong.yml
    ports:
      - "8000:8000"
      - "8001:8001"
    environment:
      KONG_DATABASE: "off"
      KONG_DECLARATIVE_CONFIG: /usr/local/kong/declarative/kong.yml
      KONG_PROXY_ACCESS_LOG: /dev/stdout
      KONG_ADMIN_ACCESS_LOG: /dev/stdout
      KONG_PROXY_ERROR_LOG: /dev/stderr
      KONG_ADMIN_ERROR_LOG: /dev/stderr
      KONG_ADMIN_LISTEN: 0.0.0.0:8001
      KONG_LOG_LEVEL: error
    networks:
      - infrastructure-network

  postgres:
    image: postgres:16-alpine
    container_name: postgres
    restart: unless-stopped
    env_file:
      - ./db/.env.dev
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./db/init:/docker-entrypoint-initdb.d
    networks:
      - infrastructure-network
    healthcheck:
      test:
        ["CMD-SHELL", "pg_isready -U infrastructure_user -d infrastructure_db"]
      interval: 10s
      timeout: 5s
      retries: 5

  auth-service:
    build:
      context: .
      dockerfile: services/auth/Dockerfile.dev
    container_name: auth-service
    restart: unless-stopped
    env_file:
      - ./services/auth/.env.dev
    volumes:
      - ./services:/app/services
    depends_on:
      postgres:
        condition: service_healthy
    networks:
      - infrastructure-network

volumes:
  postgres_data:
  frontend_node_modules:

networks:
  infrastructure-network:
    driver: bridge
